### PR TITLE
Add output warning and runtime condition when files are skipped due to size

### PR DESCRIPTION
**Pull request by michaelcfanning**
@shaopeng-gh, do you have time to add tests and finish this PR?

Currently, our driver tools silently fail when skipping a file due to size limits. We need to warn users of this circumstance.

Please make sure these warnings end up in logs, as tool configuration warnings.

### CODE REVIEW

**Comment by shaopeng-gh** [comment]
> @shaopeng-gh, do you have time to add tests and finish this PR?
> 
> Currently, our driver tools silently fail when skipping a file due to size limits. We need to warn users of this circumstance.
> 
> Please make sure these warnings end up in logs, as tool configuration warnings.

@michaelcfanning sure,

File: src/Sarif.Driver/Sdk/MultithreadedAnalyzeCommandBase.cs [hunk 1]
```diff
@@ -439,6 +445,11 @@
 
             _hashChannel.Writer.Complete();
 
+            if (_ignoredFilesCount > 0)
```
Comment (shaopeng-gh) [inline src/Sarif.Driver/Sdk/MultithreadedAnalyzeCommandBase.cs:448]: >Warnings.LogOneOrMoreFilesSkippedDueToSize(rootContext); [](http://example.com/codeflow?start=16&length=56)

@michaelcfanning question, do you think we should also change the same in file `AnalyzeCommandBase.cs` ?
If so I can work on the change together.

**Comment by shaopeng-gh** [comment]
Added a test following current pattern.

---
In reply to: [1248354509](https://github.com/microsoft/sarif-sdk/pull/2546#issuecomment-1248354509) [](http://example.com/codeflow?ancestors=1248354509,1248354509)

File: src/Sarif.Driver/Sdk/MultithreadedAnalyzeCommandBase.cs (hunk 1 above)
Comment (michaelcfanning) [reply src/Sarif.Driver/Sdk/MultithreadedAnalyzeCommandBase.cs:448]: For sure if we're missing this elsewhere, let's fix it. Here's a release note entry:

FEATURE: provide explicit logging (and new runtime condition flag) when any file is skipped due to exceeding file size limits.

Comment (shaopeng-gh) [reply src/Sarif.Driver/Sdk/MultithreadedAnalyzeCommandBase.cs:448]: Got it, working on it.

Comment (shaopeng-gh) [reply src/Sarif.Driver/Sdk/MultithreadedAnalyzeCommandBase.cs:448]: Synced with Mary, working on that internal BinSkim rule request first, will return to this once PR is published.

**Review by shaopeng-gh — APPROVED** [review]
:shipit:

**Pull request merged**